aws-xray-auto-instrumentation-agent-runtime/src/main/java/com/amazonaws/xray/agent/AgentRuntimeLoader.java

```diff
@@ -60,26 +73,60 @@ protected XRayListener xrayListenerGenerator() {
     }
 
     /**
-     * Helper method to configure the internal global recorder. It takes care to make sure
-     * no exceptions are thrown so that customer code is not impacted.
+     * Helper method to configure the internal global recorder. It can throw exceptions to interrupt customer's
+     * code at startup to notify them of invalid configuration rather than assuming defaults which might be unexpected.
      */
-    private static void configureXRay() {
-        LOG.info("Using the Centralized Sampling Strategy");
+    private static boolean configureXRay() {
         XRaySDKConfiguration configuration = XRaySDKConfiguration.getInstance();
 
-        // Configure X-Ray
-        SegmentContextResolverChain segmentContextResolverChain = new SegmentContextResolverChain();
-        segmentContextResolverChain.addResolver(new LambdaSegmentContextResolver());
-        segmentContextResolverChain.addResolver(new XRayTransactionContextResolver());
-        configuration.setSamplingStrategy(new CentralizedSamplingStrategy());
-        configuration.setSegmentContextResolverChain(segmentContextResolverChain);
-        configuration.setContextMissingStrategy(new DefaultContextMissingStrategy());
-
-        try {
+        URL configFile = getConfigFile();
+        if (configFile != null) {
+            configuration.init(configFile);
+        } else {
             configuration.init();
-        } catch(Throwable t) {
-            // If there's an exception, we just use some default
-            LOG.error("Unable to configure global recorder: " + t.getMessage());
         }
+
+        return configuration.isEnabled();
+    }
+
+    /**
+     * Helper method to retrieve the xray-agent config file's URL
+     * @return the URL of config file or null if it wasn't found
+     */
+    static URL getConfigFile() {
```

Comment:
Could you double check all the package private methods on whether you can add `// Visible for testing`? For context, the comment helps a lot when doing code review since then don't have to worry about the API, thread-safety, etc of these methods they're effectively private.

Reply:
Added it to all such methods in affected files.